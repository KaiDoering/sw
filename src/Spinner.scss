$saber-color: #03e9f4;

.light-saber {
  position: relative;
  top: 50vh;
  margin: auto;
  border-radius: 10px;
  width: 15rem;
  height: 1rem;
  box-shadow: 0 0 5px $saber-color, 0 0 25px $saber-color, 0 0 50px $saber-color, 0 0 100px $saber-color;
  transform-origin: calc(50% - 1.25rem) center;
  background: #fff;
}

.handle {
  position: relative;
  left: -2.5rem;
  bottom: 0.1rem;
  width: 3rem;
  height: 1.2rem;
  background: #7f7f7f;
}

@media (prefers-reduced-motion: no-preference) {
  .light-saber {
    animation: app-logo-spin infinite 4s linear;
  }
}

@keyframes app-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
