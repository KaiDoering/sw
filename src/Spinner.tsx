import './Spinner.scss';

import React from 'react';

function Spinner() {
  return <div className="light-saber"><div className="handle" /></div>;
}

export default Spinner;
