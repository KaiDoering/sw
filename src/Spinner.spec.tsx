import React from 'react';

import { render } from '@testing-library/react';

import Spinner from './Spinner';

describe('<Spinner />', () => {
  test('renders correctly', () => {
    const page = render(<Spinner />);

    expect(page.asFragment()).toMatchSnapshot();
  });
});
