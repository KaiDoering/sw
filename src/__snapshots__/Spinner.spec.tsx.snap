// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="light-saber"
  >
    <div
      class="handle"
    />
  </div>
</DocumentFragment>
`;
